rasa_core/training/interactive.py

```diff
@@ -1060,21 +1069,46 @@ def _validate_nlu(intents: List[Text],
 
 def _correct_entities(latest_message: Dict[Text, Any],
                       endpoint: EndpointConfig,
-                      sender_id: Text) -> Dict[Text, Any]:
+                      sender_id: Text) -> [Dict[Text, Any]]:
     """Validate the entities of a user message.
 
     Returns the corrected entities"""
     from rasa_nlu.training_data.formats import MarkdownReader
 
-    entity_str = _as_md_message(latest_message.get("parse_data", {}))
+    parse_data = latest_message.get("parse_data", {})
+    entity_str = _as_md_message(parse_data)
     question = questionary.text(
         "Please mark the entities using [value](type) notation",
         default=entity_str)
 
     annotation = _ask_or_abort(question, sender_id, endpoint)
     # noinspection PyProtectedMember
     parsed = MarkdownReader()._parse_training_example(annotation)
-    return parsed.get("entities", [])
+    if entity_str == parsed:  # user did not change annotation
+        return parse_data.get("entities", [])
+
+    else:  # update annotated entities
+        entities = []
```

Comment:
This code is way to hard to read. I tried to simplify it, and this should roughly do the same thing, right?

```python
        entities = parsed.get("entities", [])[:]

        # overwrite entities which have already been 
        # annotated in the original annotation to preserve
        # additional entity parser information
        for i, entity in enumerate(entities):
            for original_entity in parse_data.get("entities", []):
                if _is_same_entity_annotation(entity, original_entity):
                    entities[i] = original_entity
                    break
        return entities

# adding this function as well
def _is_same_entity_annotation(entity, other):
    return (entity['value'] == other['value'] and
            entity['entity'] == other['entity'])

```
I also don't think we need the if else in L1087 (it should be fine to always run the above, even if the user did not change anything)